plaso/storage/interface.py

```diff
@@ -4,172 +4,237 @@
 from __future__ import unicode_literals
 
 import abc
-import os
-import shutil
-import tempfile
 
+from plaso.containers import artifacts
+from plaso.containers import event_sources
+from plaso.containers import events
+from plaso.containers import reports
+from plaso.containers import sessions
+from plaso.containers import tasks
+from plaso.containers import warnings
 from plaso.lib import definitions
 from plaso.serializer import json_serializer
 
 
-class SerializedAttributeContainerList(object):
-  """Serialized attribute container list.
-
-  The list is unsorted and pops attribute containers in the same order as
-  pushed to preserve order.
-
-  The GetAttributeContainerByIndex method should be used to read attribute
-  containers from the list while it being filled.
+# pylint: disable=redundant-returns-doc,redundant-yields-doc
+class BaseStore(object):
+  """Storage interface.
 
   Attributes:
-    data_size (int): total data size of the serialized attribute containers
-        on the list.
-    next_sequence_number (int): next attribute container sequence number.
+    format_version (int): storage format version.
+    serialization_format (str): serialization format.
+    storage_type (str): storage type.
```

Comment:
what is the rational to make the formats part of the interface?

Reply:
These haven't moved, hopefully the cleaner diff makes this easier to see.

Reply:
ack, staged changes help in clarification